### DISCUSSION TITLE
Why Github has not listed "Pakistan" in the country select list for 2FA?

### DISCUSSION THREAD

**Discussion by bilaldotdev** [community/community#109129] [Other Feature Feedback, Questions, & Ideas]
### My point of view

Thousands of users from Pakistan are using github and now when github wants their user to enable 2FA,  **Pakistan** is not even listed in the country list? This is shameful and heartbreaking to see a big organization doing this type of discrimination.

![Screenshot from 2024-02-19 11-56-01](https://github.com/community/community/assets/46219560/a249b5b9-50ed-4820-b1b0-6bf0568e2926)

**Comment by queenofcorgis** [comment]
If your country is not on [the list](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported#supported-countries-for-sms-authentication), then we aren't currently able to reliably deliver text messages to your country. We track delivery rates across all countries, including ones not in the list, and review it to understand when we have to remove countries or can add them back. We do not add countries with low deliverability rates, because it leads to users being locked out of their accounts when they can’t receive the SMS for 2FA.

If GitHub doesn’t support two-factor authentication via text message for your country of residence, you must set up authentication via a TOTP application. For more information on how to configure 2FA, please review our [documentation](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication#configuring-two-factor-authentication-using-a-totp-mobile-app).

**Comment by MusawarAhmed** [comment]
While I understand that GitHub prioritizes deliverability to prevent account lockouts, excluding a country with one of the fastest-growing developer populations in the world is not a sustainable solution.

The reliance on TOTP apps is a valid technical alternative, but it ignores the local context where SMS remains a primary and most accessible form of verification for many. By not listing Pakistan, GitHub creates an unnecessary barrier for thousands of contributors who are now required to enable 2FA but feel sidelined by the platform they contribute to daily.

If deliverability is the issue, I urge GitHub to:

Partner with local telecommunications providers in Pakistan to stabilize routing.

Provide transparency on what specific delivery benchmarks need to be met for a country to be "re-listed."

Consider WhatsApp or alternative localized API integrations that have much higher success rates in the region than traditional SMS.

We hope to see GitHub take proactive steps to include the Pakistani developer community rather than simply directing us to documentation.

**Comment by bilaldotdev** [comment]
many many and many products are able to support SMS based OPT in Pakistan without any problem. The issue is not "reliability", it's priorities of github and they clearly don't have a reason to consider Pakistan.

**Comment by Arpon-09** [comment]
On GitHub, SMS 2FA is only supported in countries where SMS delivery is reliable.

GitHub documentation says they only support SMS 2FA for certain countries based on SMS delivery success rates. If delivery is unreliable, they do not include that country because users might get locked out of their accounts.

**Comment by krillmango** [comment]
TOTP does not compromise privacy

**Comment by code-with-bhumi** [comment]
Hello @bilaldotdev 
> From GitHub’s perspective, the exclusion is not an "issue" or bug, but a deliberate security measure to protect you from being locked out of your account

Pakistan is excluded from SMS-based two-factor authentication due to **low message delivery success rates**, posing a risk of account lockout. To ensure reliability, GitHub limits SMS 2FA to regions with high carrier performance and recommends TOTP authenticator apps for unsupported areas. 

- Read the full policy details at [GitHub Docs](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported)

  <img width="1269" height="1592" alt="Countries-where-SMS-authentication-is-supported-GitHub-Docs-03-23-2026_04_11_PM" src="https://github.com/user-attachments/assets/0ad09490-3d43-4fac-bd37-dec893c905aa" />

**Comment by krillmango** [comment]
https://totp.app

**Comment by krillmango** [comment]
https://twilio.com/en-us/guidelines/pk/sms

**Comment by 8171webportals** [comment]
I can understand why users in Pakistan would feel frustrated about this, especially since many developers, students, and professionals rely on GitHub daily for coding, collaboration, and open-source contributions. When 2FA becomes mandatory but SMS support is unavailable for certain countries, it can create the feeling that users are being left out of important security features.

At the same time, GitHub usually recommends authenticator apps or security keys instead of SMS because SMS-based verification is considered less secure and sometimes unreliable in different regions. Still, better communication and broader country support would definitely help reduce confusion and frustration for affected users.

A lot of Pakistani developers are active contributors in the global tech community, so accessibility and equal support for security features should be improved wherever possible.

I also came across some discussions on https://8171web-portals.pk/ about digital accessibility and how online platforms can improve support systems for users across different regions and communities.

Overall, providing more inclusive authentication options and clearer regional support policies would create a much better experience for international users.

**Comment by omanomer53-lab** [comment]
I can understand why this would be frustrating, especially for users who rely on SMS for two-factor authentication and suddenly find that their country isn’t available in the selection list. The discussion also highlights how important it is for major platforms to make security features accessible across different regions rather than assuming every user has the same verification options. Hopefully issues like this get more attention as more people report them. [BISE D.G. Khan 12th class result 2026](https://12classresults.com/)
